src/reducers/waypoint-reducer.ts

```diff
@@ -54,19 +54,27 @@ export const {
 /**
  * Checks wheter route ID has been initialized. If so, gets location and MNC code and creates
  * a new waypoint object which will stored into localdevices `WaypointState`
- * Sends also waypoints to the server, if there are more than 5 pending waypoints
+ * Sends waypoints to server according to the sending interval or immediately if
+ * was disconnected and is connected.
  * @returns dispatch method to update `WaypointState`
  */
+let wasOffline = false;
+let sendTicker = 0;
 export const storeAndSendWaypoints = () => {
 	return async (dispatch: AppDispatch, getState: () => ReduxState) => {
 		const routeId = getState().waypoints.routeId;
 		const pendingWaypoints = getState().waypoints.pendingWaypoints;
+		const sendingInterval = getState().user.sendingInterval;
+		const trackingInterval = getState().user.trackingInterval;
+		const location = await Location.getLastKnownPositionAsync({});
+		const networkCode = await Cellular.getMobileNetworkCodeAsync();
+		const netInfo = getNetworkCellularGeneration(
+			await NetworkConnectionInformation()
+		);
+		const isConnected = await (
+			await NetworkConnectionInformation()
+		).isConnected;
```

Comment:
The first await seems to be pointless here? Use instead `(await NetworkConnectionInformation()).isConnected;`?